buildSrc/src/main/java/org/elasticsearch/gradle/transform/UnpackTransform.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.transform;
+
+import org.gradle.api.artifacts.transform.CacheableTransform;
+import org.gradle.api.artifacts.transform.InputArtifact;
+import org.gradle.api.artifacts.transform.TransformAction;
+import org.gradle.api.artifacts.transform.TransformOutputs;
+import org.gradle.api.artifacts.transform.TransformParameters;
+import org.gradle.api.file.FileSystemLocation;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.internal.UncheckedException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@CacheableTransform
+public interface UnpackTransform extends TransformAction<TransformParameters.None> {
+
+    @PathSensitive(PathSensitivity.NAME_ONLY)
+    @InputArtifact
+    Provider<FileSystemLocation> getArchiveFile();
+
+    @Override
+    default void transform(TransformOutputs outputs) {
+        File archiveFile = getArchiveFile().get().getAsFile();
+        String unpackedDirName = removeExtension(archiveFile.getName());
+        File unzipDir = outputs.dir(unpackedDirName);
+        try {
+            unpack(archiveFile, unzipDir);
+        } catch (IOException e) {
+            throw UncheckedException.throwAsUncheckedException(e);
+        }
+    }
+
+    void unpack(File tarFile, File targetDir) throws IOException;
```

Comment:
This argument should probably be called `archiveFIle`.

Reply:
fixed